lib/Sema/TypeCheckPattern.cpp

```diff
@@ -19,16 +19,83 @@
 #include "TypeCheckAvailability.h"
 #include "TypeCheckType.h"
 #include "swift/Basic/StringExtras.h"
-#include "swift/AST/ASTWalker.h"
 #include "swift/AST/ASTVisitor.h"
-#include "swift/AST/SourceFile.h"
+#include "swift/AST/ASTWalker.h"
 #include "swift/AST/NameLookup.h"
+#include "swift/AST/SourceFile.h"
 #include "swift/AST/ParameterList.h"
 #include "swift/AST/TypeCheckRequests.h"
 #include "llvm/Support/SaveAndRestore.h"
 #include <utility>
 using namespace swift;
 
+llvm::SmallVector<Identifier, 4> extractArgsFromPattern(ASTContext &Context,
+                                                        Pattern *pattern) {
+  auto &diags = Context.Diags;
+  llvm::SmallVector<Identifier, 4> params;
+  // Get Pattern kind
+  switch (pattern->getKind()) {
+  case PatternKind::Tuple: {
+    // Walk through tuple to get argument names
+    TuplePattern *tp = cast<TuplePattern>(pattern);
+    Optional<bool> usingLabels = None;
+    for (auto elem : tp->getElements()) {
+      Identifier label = elem.getLabel();
+      if (label.empty()) {
+        if (elem.getPattern()->getKind() == PatternKind::Any)
+          params.push_back(Identifier());
+        else
+            // Set None if not using labels, throw error if contradicts
+            if (usingLabels == None)
```

Comment:
This if-else tree and the one below it absolutely positively need explicit braces.  I can't track anything here.